sympy/solvers/solveset.py

```diff
@@ -1111,40 +1070,47 @@ def _is_exponential(f, symbol):
     Returns
     =======
 
-    `True` if the equation is in exponential form otherwise `False`.
+    `True` if the equation is exactly two term exponential form
+    otherwise `False`.
 
     Examples
     ========
 
     >>> from sympy import symbols, cos, exp
     >>> from sympy.solvers.solveset import _is_exponential as check
     >>> x, y = symbols('x y')
-    >>> check(x**y - x, y)
-    True
+    >>> check(x**y - x**2, y)
+    False
     >>> check(x**y - x, x)
     False
-    >>> check(exp(x + 3), x)
+    >>> check(exp(x + 3) + 3**x, x)
     True
     >>> check(cos(2**x), x)
     False
 
     * Philosophy behind the helper
 
-    The function extracts each term of the equation and checks if it is
-    of exponential form w.r.t `symbol`.
+    The function checks for all two terms being in power form
+    with `symbol` in their exponents.
     """
 
-    expr_args = _term_factors(f)
-    for expr_arg in expr_args:
-        if isinstance(expr_arg, (Pow, exp)) and (
-                symbol in expr_arg.exp.free_symbols):
-            return True
-    return False
+    if not (isinstance(f, Add) and len(f.args) == 2):
```

Comment:
Changes have been made to make `_is_exponenial` return `True` only for which `_solve_exponential` could solve. Previously those equations returned `True` that might not get solved by `_solve_exponential` thereby causing infinite recursion.

Now the routine itself return `False` if there are no two terms in the expression and further checks for all terms being exponent w.r.t `symbol`, if so only then it would return `True` otherwise `False`. This condition is mainly to counter expressions like `x**2 - 2**x` or `exp(x) - x`, which are mostly lambert forms